src/audio/module_adapter/module_adapter_ipc4.c

```diff
@@ -67,15 +67,20 @@ module_ext_init_decode(struct comp_dev *dev, struct module_config *dst,
 				(const struct ipc4_module_init_ext_obj_dp_data *)(obj + 1);
 
 			if (obj->object_words * sizeof(uint32_t) < sizeof(*dp_data)) {
-				comp_err(dev, "dp_data object too small %zu < %zu",
-					 obj->object_words * sizeof(uint32_t), sizeof(*dp_data));
-				return NULL;
+				comp_warn(dev, "dp_data object too small %zu < %zu",
+					  obj->object_words * sizeof(uint32_t), sizeof(*dp_data));
+				break;
```

Comment:
why is this not fatal any more? Also `break` only takes you out of `switch`, not out of the loop, is that correct?

Reply:
This is mandatory if we go and redefine "struct ipc4_module_init_ext_obj_dp_data". Otherwise payloads sent by an older linux version would always fail dp module creation.

But fear not, non of the data in the too short data object is used. Its all ignored as all of that code accessing the data is within the case block. And we need to continue the decoding loop in case there is some other objects there that we can decode correctly.